src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs

```diff
@@ -170,5 +171,26 @@ public static bool IsReferenceOfType(string reference, Dictionary<string, Schema
 
             return false;
         }
+
+        /// <summary>
+        /// Returns array of resource providers
+        /// </summary>
+        /// <param name="paths">Dictionary of paths to look for</param>
+        /// <returns>Array of resource providers</returns>
+        public static string[] GetResourceProviders(Dictionary<string, Dictionary<string, Operation>> paths)
+        {
+            if (paths == null)
+            {
+                throw new System.ArgumentNullException(nameof(paths));
```

Comment:
Is someone catching this exception at a higher level in the stack or is this method not being invoked when 'paths' is null, I'm wondering because we may have specs that don't have paths (like the ones we're starting to see with just model definitions) and I wouldn't want the rule to fail for this.